wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```
@@ -547,6 +540,25 @@ private boolean checkPreconditionNotEmpty(Boolean isConditionSQL) {
     return false;
   }
 
+  private boolean checkSQLExecution(Config config) {
+    if (!(Feature.WRANGLER_PRECONDITION_SQL.isEnabled(getContext())
+            || Feature.WRANGLER_EXECUTION_SQL.isEnabled(getContext()))) {
+      // disabling SQL execution for precondition and directives
+      return false;
+    }
+
+    if (!Strings.isNullOrEmpty(config.getPreconditionLanguage())) {
+      return SQL.equalsIgnoreCase(config.getPreconditionLanguage());
+    }
+
+    if (!Strings.isNullOrEmpty(config.getSqlExecution())) {
```

We are enabling SQL only in certain directives and not all. Where is that condition checked? Even when the SQL mode is enabled, If the given list of directives cant be supported in SQL, we should revert back to old way of execution.

---

In that case if the directive is not supported, it should return an error and normal flow should work.

---

Is that tested?

---

Will add changes to validate supported directives in the next PR as it requires some changes to the directive interface. For now if the directive is not supported the normal execution takes place.